### ISSUE TITLE
Building glew without GLX

### ISSUE THREAD

**Issue by vanfanel** [nigels-com/glew#328]
Hi there,

Since more and more systems are jumping to Wayland (and embedded GNU/Linux systems use KMS/DRM or Wayland), GLX is becoming something of the past.
There's GLVND, which is vendor-gnostic, for full OpenGL without X11/GLX, too.

So, is it possible to build GLEW without X11/GLX? How?
Looking at the docs, I see `SYSTEM=linux-egl` but it seems to require GLX too...

**Comment by cbix** [comment]
GLFW [is implementing runtime platform selection](https://github.com/glfw/glfw/blob/master/docs/intro.dox#L159) for the upcoming version and by default it's using the first platform that works, so existing software keeps working. It's a slightly different focus but it would be great to see a similar behavior in GLEW wrt. GLX/EGL if compiled with both.

**Comment by nigels-com** [comment]
Yes, I agree it makes sense for things to move in that direction.
I'm not sure GLEW can do that while remaining compatible, but I agree it should be looked at.

**Comment by chewi** [comment]
I maintain the glew package on Gentoo Linux. I had a user suggest we support `linux-egl` as well as `linux`, which we can do as a source-based distribution, but having it not support X11 and native Wayland at the same time is slightly irritating. I don't understand this stuff very well, but I thought that X11 supported EGL anyway? I'm surprised that doesn't just work.

```
env -u WAYLAND_DISPLAY supertux2 --renderer opengl
[FATAL] SuperTux-v0.6.3-Source/src/supertux/main.cpp:672 Unexpected exception: GLVideoSystem: GlewError: Missing GL version

env -u WAYLAND_DISPLAY x64dtv.gtk                 
x64dtv.gtk: symbol lookup error: x64dtv.gtk: undefined symbol: __glewXSwapIntervalSGI
```

**Comment by nigels-com** [comment]
It's a bit of a historical problem, on the face of it.  GLEW was compiled WGL-mode on Windows, and GLX-mode on not-Windows.
With the addition of EGL we have the choice to compile in EGL-mode for either platform, but that's not a runtime choice currently, it's a compile-time one.  I'm not sure it _can_ be made a runtime choice in a compatible sort of way.

**Comment by chewi** [comment]
Okay, but that doesn't quite address my point. If I build it for EGL only, why doesn't it work under X11? It's definitely a thing, [Firefox can do it](https://mastransky.wordpress.com/2021/10/30/firefox-94-comes-with-egl-on-x11/), for example. I'm guessing it needs some additional code in GLEW, but it sounds like it would effectively solve this problem?

**Comment by chewi** [comment]
Maybe it would be a lot of work, but I wonder whether [Waffle](https://waffle.freedesktop.org) could help here. I heard about it at FOSDEM a few years ago.

**Comment by nigels-com** [comment]
The reluctance is about breaking compatibility with existing applications that are expecting GLX.  The solution would likely be a libGLEW3.so for supporting either/or GLX and EGL on Linux, WGL and EGL on Windows.
